src/output/plugins/SndioOutputPlugin.cxx

```diff
@@ -163,9 +158,34 @@ SndioOutput::Play(const void *chunk, size_t size)
 	return n;
 }
 
+void
+SndioOutput::SetVolume(unsigned int volume) {
+	sio_setvol(sio_hdl, volume * SIO_MAXVOL / 100);
+}
+
+void
+SndioOutput::VolumeChanged(unsigned int _raw_volume) {
+	raw_volume = _raw_volume;
+	if (listener != NULL && mixer != NULL) {
+		listener->OnMixerVolumeChanged(*mixer,
+		    _raw_volume * 100 / SIO_MAXVOL);
+	}
+}
+
+unsigned int
+SndioOutput::GetVolume() {
+	return (raw_volume * 100 / SIO_MAXVOL);
```

Comment:
Duplicate formula; this code is already present in `VolumeChanged()`. Maybe better remember the MPD-style value instead of the raw value?

Reply:
I thought about this, but honestly don't remember the reason right now. I believe the code would be duplicated elsewhere when storing the mpd value, or rounding errors would appear when a third party changes the volume.

Reply:
OK, move the formula to a small function. Don't duplicate code. This is maintenance hell.